optimum/intel/generation/modeling.py

```diff
@@ -116,6 +118,7 @@ def __init__(
         self._device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
         self.normalized_config = NormalizedConfigManager.get_normalized_config_class(config.model_type)(config)
         self.model_dtype = kwargs.get("model_dtype", None)
+        self.has_position_ids = has_position_ids

```

Comment:
no need to have an attribute, we can use  [`MODEL_TYPES_REQUIRING_POSITION_IDS`](https://github.com/huggingface/optimum/blob/e7bd60dd2c1e295263ba57a4e468a62ab5b179e8/optimum/exporters/onnx/utils.py#L69)  directly

Reply:
Thanks, I will use it.